src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt

```diff
@@ -120,33 +120,37 @@ class SnykLanguageClient :
 
         //If the diagnostics for the file is empty, clear the cache.
         if (firstDiagnostic == null) {
-            scanPublisher.onPublishDiagnostics("code", snykFile, emptyList())
-            scanPublisher.onPublishDiagnostics("oss", snykFile, emptyList())
+            clearCache(scanPublisher, snykFile)
             return
         }
 
+        val issueList = getScanIssues(diagnosticsParams)
+        updateCacheForProduct(scanPublisher, product, snykFile, issueList)
+
+        return
+    }
+
+    fun getScanIssues(diagnosticsParams: PublishDiagnosticsParams): List<ScanIssue> {
         val issueList = diagnosticsParams.diagnostics.stream().map {
             Gson().fromJson(it.data.toString(), ScanIssue::class.java)
         }.toList()
+        return issueList
+    }
 
-        when (product) {
-            LsProductConstants.OpenSource.value -> {
-                scanPublisher.onPublishDiagnostics(product, snykFile, issueList)
-            }
-
-            LsProductConstants.Code.value -> {
-                scanPublisher.onPublishDiagnostics(product, snykFile, issueList)
-            }
-
-            LsProductConstants.InfrastructureAsCode.value -> {
-                // TODO implement
-            }
+    fun clearCache(scanPublisher: SnykScanListenerLS, snykFile: SnykFile) {
+        updateCacheForProduct(scanPublisher, "code", snykFile, emptyList())
+        updateCacheForProduct(scanPublisher, "oss", snykFile, emptyList())
+    }
 
-            LsProductConstants.Container.value -> {
-                // TODO implement
-            }
+    private fun updateCacheForProduct(
```

Comment:
I wouldn't extract this to a method for effectively 3 lines of code. The effect may be updating the cache, but the event bus usage makes it flexible - there could be other listeners. I would call it something like, well, publishDiagnostics - and then the question is, if extracting it really gives us benefits in maintainability and understandability.